docs/week-3.md

```diff
@@ -0,0 +1,523 @@
+
+    return(
+        <div>
+            <h1>My Lucky Number is {myLuckyNumber}</h1>
+        </div>
+    );
+}
+
+export default LuckyNum;
+```
+
+Now that we have access to `props` we can simply display the luckyNum field of the prop object.
+
+```
+import React from 'react'; 
+
+const LuckyNum = (props) => (
+
+    <div>
+        <h1>My Lucky Number is {props.luckyNumber}</h1>
+    </div>
+
+)
+
+export default LuckyNum;
+```
+Our component is now reuseable!! We now can use it on Doug's profile like so.
+
+```
+import React from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is Doug! My favorite color is purple.</h1>
+        <LuckyNum luckyNum={1111}/>
+    </div>
+);
+
+export default Profile;
+```
+
+`props` is an object that React makes for each component. When you pass parameters into component declaration 
+like we did with `LuckNum` React adds the parameter as a field of the object `props`. In order to get the field
+of the object we simply access the object's field like we would any other object. 
+
+> We can make the `Profile` component more reusable with props. Try now to add props to make `Profile` more
+>generic (possible answer below). 
+
+```
+import React from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = (props) => (
+
+    <div>
+        <img src=`${props.profileURL}`/>
+        <h1>Hi, my name is {props.name}! My favorite color is {props.favoriteColor}.</h1>
+        <LuckyNum luckyNum={props.luckyNum}/>
+    </div>
+);
+
+export default Profile;
+```
+
+## State Management
+Alright. So John is now super happy with his profile and wants to see if other people like his profile as much
+as he does. To do this John adds a like button on his profile. 
+
+```
+import React from 'react';
+import LuckyNum from './LuckyNum';
+
+const Profile = () => (
+
+    <div>
+        <img src="profileImage.jpg"/>
+        <h1>Hi, my name is John! My favorite color is blue.</h1>
+        <LuckyNum />
+        <button>Likes: </button>
+    </div>
+);
+
+export default Profile;
+```
+
+John is now perplexed. How can I keep track of the number of likes and how can I keep the number consistent
+through each render? Welcome to state management John.
+
+### What is managing state?
+You can think of state as the data that persist through each clock render of your application. Under the hood,
+your react application renders the screen on clock ticks. In order to make your non-constant data persist we
+have to manage the state of our application. We manage the state in our componetns using the react hook 
+`useState`. There are other ways of managing state in react, but this is a simple and easy way.
+
+### useState Hook
+`useState` allows us to add and keep track of data in the state. In order to use this hook we must import 
+it from react. We use {} when importing useState because it is not a default export but a named one. We will
```

Comment:
```suggestion
it from React. We use {} when importing `useState` because it is a named export. This is different from a default export, which is how we were getting exports earlier. We will
```

or something?